gateway/test/dapi_alpha_test.go

```diff
@@ -11,3 +14,757 @@ func (s *GatewayOpsTestSuite) TestDapiAlphaEnabled() {
 	})
 	requireRestSuccess(s.T(), resp)
 }
+
+func (s *GatewayOpsTestSuite) insertBinaryDocument() string {
+	docId := s.randomDocId()
+	resp := s.sendTestHttpRequest(&testHttpRequest{
+		Method: http.MethodPost,
+		Path: fmt.Sprintf(
+			"/v1.alpha/buckets/%s/scopes/%s/collections/%s/documents/%s/increment",
+			s.bucketName, s.scopeName, s.collectionName, docId,
+		),
+		Headers: map[string]string{
+			"Authorization": s.basicRestCreds,
+		},
+		Body: TEST_BINARY_CONTENT,
+	})
+	requireRestSuccess(s.T(), resp)
+	assertRestValidEtag(s.T(), resp)
+	assertRestValidMutationToken(s.T(), resp, s.bucketName)
+
+	return docId
+}
+
+func (s *GatewayOpsTestSuite) TestAppendToDocument() {
+	s.Run("Basic", func() {
+		docId := s.insertBinaryDocument()
+
+		// Append to the document
+		resp := s.sendTestHttpRequest(&testHttpRequest{
+			Method: http.MethodPost,
+			Path: fmt.Sprintf(
+				"/v1.alpha/buckets/%s/scopes/%s/collections/%s/documents/%s/append",
+				s.bucketName, s.scopeName, s.collectionName, docId,
+			),
+			Headers: map[string]string{
+				"Authorization": s.basicRestCreds,
+			},
+			Body: TEST_APPEND_CONTENT,
+		})
+		requireRestSuccess(s.T(), resp)
+		assertRestValidEtag(s.T(), resp)
+		assertRestValidMutationToken(s.T(), resp, s.bucketName)
+
+		// Verify the key-value is present
+		resp = s.sendTestHttpRequest(&testHttpRequest{
+			Method: http.MethodGet,
+			Path: fmt.Sprintf(
+				"/v1/buckets/%s/scopes/%s/collections/%s/documents/%s",
+				s.bucketName, s.scopeName, s.collectionName, docId,
+			),
+			Headers: map[string]string{
+				"Authorization": s.basicRestCreds,
+			},
+		})
+		requireRestSuccess(s.T(), resp)
+		println("Response body is: ", string(resp.Body))
+		assert.Contains(s.T(), string(resp.Body), string(TEST_APPEND_CONTENT))
+	})
+
+	s.Run("Forbidden", func() {
```

Comment:
This test appears to be testing an output of the endpoint as opposed to testing the specific case.  I believe there are a couple of cases which will lead to Forbidden such as InvalidAuth, Unauthenticated and PermissionDenied.  See dapi_crud_test.go for examples of these.  Additionally, I believe the test framework includes a mechanism for automatically running all these common failure cases.